reascripts/ReaSpeech/source/ReaSpeechUI.lua

```diff
@@ -183,6 +191,30 @@ function ReaSpeechUI:new_jobs(jobs, endpoint, callback)
   table.insert(self.requests, request)
 end
 
+function ReaSpeechUI.link(text, onclick, text_color, underline_color)
+  text_color = text_color or 0xffffffff
+  underline_color = underline_color or 0xffffffa0
+
+  ImGui.TextColored(ctx, text_color, text)
+
+  if ImGui.IsItemHovered(ctx) then
+    local rect_min_x, rect_min_y = ImGui.GetItemRectMin(ctx)
+    local rect_max_x, _ = ImGui.GetItemRectMax(ctx)
+    local _, rect_size_y = ImGui.GetItemRectSize(ctx)
+    local line_y = rect_min_y + rect_size_y - 1
+
+    ImGui.DrawList_AddLine(
+      ImGui.GetWindowDrawList(ctx),
+      rect_min_x, line_y, rect_max_x, line_y,
+      underline_color, 1.0)
+    ImGui.SetMouseCursor(ctx, ImGui.MouseCursor_Hand())
+  end
+
+  if ImGui.IsItemClicked(ctx) then
+    onclick()
+  end
+end
+
 function ReaSpeechUI.png_from_bytes(image_key)
```

Comment:
this whole thing too, for what it's worth (re: where this code lives in the codebase). my hesitation on this was maybe preferring something with a use api like `IMAGES.{image_key}.png()` 🤔

Reply:
I moved these to a new "Widgets" module. We can always revisit later, but it seems cleaner.